esp-hal/src/i2c.rs

```diff
@@ -1463,86 +1511,120 @@ pub trait Instance: crate::private::Sealed {
     }
 
     /// Configures the I2C peripheral for a write operation.
-    fn setup_write<'a, I>(&self, addr: u8, bytes: &[u8], cmd_iterator: &mut I) -> Result<(), Error>
+    /// - `addr` is the address of the slave device.
+    /// - `bytes` is the data two be sent.
+    /// - `start` indicates whether the operation should start by a START
+    ///   condition and sending the address.
+    /// - `cmd_iterator` is an iterator over the command registers.
+    fn setup_write<'a, I>(
+        &self,
+        addr: u8,
+        bytes: &[u8],
+        start: bool,
+        cmd_iterator: &mut I,
+    ) -> Result<(), Error>
     where
         I: Iterator<Item = &'a COMD>,
     {
-        if bytes.len() > 254 {
+        let max_len = if start { 254usize } else { 255usize };
```

Comment:
I don't think I understand why we need the +1 byte.

Reply:
Could you add a comment? Just so future us don't have to come back to the PR :D